packages/clerk-js/src/ui/elements/ArrowBlockButton.tsx

```diff
@@ -7,9 +7,16 @@ import { ArrowRightIcon } from '../icons';
 import type { PropsOfComponent, ThemableCssProp } from '../styledSystem';
 
 type ArrowBlockButtonProps = PropsOfComponent<typeof Button> & {
+  /**
+   * @deprecated use leftIcon and leftIconSx instead
```

Comment:
How come we deprecated an internal prop instead of refactoring its usages at the same time?

Reply:
@nikosdouvlis  

I tried to implement this based on this comment
```
  // TODO: Replace this with a simple Icon prop like left icon
  // This was implemented this way to allow for setting its descriptor
  // from the outside, but we can use `leftIconDescriptor` pattern instead
  ```
  
  I later found out that in many different part of the code base we pass an `<image/>` as an icon. And I feel that handling that as well it would be outside the scope of this PR. 
  
  How do you want me to proceed this ? Should I revert the last commit that introduces the `leftIcon` prop ?

Reply:
Makes sense, lets do it on a 2nd PR